yt/yt/core/http/retrying_client.cpp

```diff
@@ -18,14 +19,46 @@ static const auto& Logger = HttpLogger;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-class TJsonResponseChecker
+class TResponseCheckerBase
     : public IResponseChecker
 {
+public:
+    TResponseCheckerBase(TRetryChecker retryChecker = BIND(&DefaultRetryChecker))
+        : RetryChecker_(std::move(retryChecker))
+    { }
+
+    virtual bool IsRetriableError(const TError& error) override 
+    {
+        return RetryChecker_(error);
+    }
+
+    virtual TError CheckError(const IResponsePtr& response) override = 0;
+    virtual NYTree::INodePtr GetFormattedResponse() const override = 0;
+
+protected:
+    static bool DefaultRetryChecker(const TError& error)
+    {
+        // TODO: default retry checker
+        Y_UNUSED(error);
```

Comment:
We prefer to hide name in the method declaration, i.e:
```
static bool DefaultRetryChecker(const TError& /*error*/)
```

Reply:
This is actually not method declaration. There should be a default retry checking logic. Added in new commit.